lib/cocoapods/validator.rb

```diff
@@ -28,8 +28,12 @@ class Validator
     #         the Source URLs to use in creating a {Podfile}.
     #
     def initialize(spec_or_path, source_urls)
-      @source_urls = source_urls.map { |url| config.sources_manager.source_with_name_or_url(url) }.map(&:url)
       @linter = Specification::Linter.new(spec_or_path)
+      @source_urls = if !@linter.spec || @linter.spec.dependencies.empty?
```

Comment:
Hm. I see. Would it make sense to check `repo_update` in the `sources` method in `analyzer.rb`? If repo_update is false, just return `podfile.sources`.